bundle/manifests/aws-load-balancer-operator.clusterserviceversion.yaml

```diff
@@ -249,7 +249,8 @@ spec:
                 - --upstream=http://127.0.0.1:8080/
                 - --logtostderr=true
                 - --v=0
-                image: gcr.io/kubebuilder/kube-rbac-proxy:v0.8.0
+                - --http2-disable
+                image: gcr.io/kubebuilder/kube-rbac-proxy:v0.15.0
```

Comment:
I noticed external-dns-operator [uses](https://github.com/openshift/external-dns-operator/blob/b7633e786424c8209b4fe1fccb011b212ea48357/bundle/manifests/external-dns-operator.clusterserviceversion.yaml#L484) `quay.io/openshift/origin-kube-rbac-proxy:latest`.

Should we be using quay.io images instead of other external images? Also, `latest` tag vs. pinning to specific version is different.

Reply:
Will we also need a substitution like we have for external-dns-operator? https://github.com/openshift/release/blob/f4f1bf0fe4f6af1eb6525431de841db8ba9a22b8/ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-release-4.16.yaml#L36-L37

Reply:
Hm, I'm not sure what this substitution gives us. I see only the concrete OCP tag instead of `latest`. The base image is still RHEL (btw this is a good thing compared to gcr.io image as we would like to be closer to what we ship). I'm fine with the latest tag because it avoids a drift between the local testing and CI testing plus doesn't bloat the CI config.

Reply:
Hm, so external-dns-operator is using `quay.io/openshift/origin-kube-rbac-proxy:latest` despite the substitution that is specified in the CI configuration?  In that case, I'm not sure what the point of the substitution is.  I figured it would substitute in the kube-rbac-proxy image from the release payload.  Does such substitution only happen for branches other than the main branch?

Reply:
> so external-dns-operator is using quay.io/openshift/origin-kube-rbac-proxy:latest despite the substitution that is specified in the CI configuration?

There are 3 places: developer's machine where what's committed in GitHub repository is used as the image, CI where a substitution may happen before the test is run, CPaaS where the registry has to change to the redhat.registry.com.
I think that keeping these 3 cases as close as possible to each other is a goal. However some diffs are inevitable as some code is open (for contributions) and some is closed (for Red Hat products).

> I figured it would substitute in the kube-rbac-proxy image from the release payload.

Yes, I also think it's the purpose of the substitution. However I greped on the CI config and only 3 operators do the substitution for `kube-rbac-proxy` and 2 of them done by me :) 

> Does such substitution only happen for branches other than the main branch?

Well, indeed, `main` doesn't have. I [removed it](https://github.com/openshift/release/commit/23de4d9d68180c559bbc400d78476ebd0e997b7d#diff-8f9a46e5fe5d27abff924e731c5d0578023e92a751e5187512510e8bd72369eb) as part of the works to reduce the dependency on all things OpenShift. 


I don't see a reason of why we should replace kube-rbac-proxy image with the payload one. I'll do a PR to remove it to simplify the CI config and to reduce the drift between the local development and CI testing.

Reply:
Asked to opt out from the ART automation again: https://redhat-internal.slack.com/archives/CB95J6R4N/p1701723839704069.